api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1620,12 +1620,10 @@ public InputStream getObject(String bucketName, String objectName, long offset,
     }
 
     Map<String,String> headerMap = new HashMap<>();
-    if (offset > 0) {
-      if (length != null) {
-        headerMap.put("Range", "bytes=" + offset + "-" + (offset + length - 1));
-      } else {
-        headerMap.put("Range", "bytes=" + offset + "-");
-      }
+    if (length != null) {
+      headerMap.put("Range", "bytes=" + offset + "-" + (offset + length - 1));
+    } else {
+      headerMap.put("Range", "bytes=" + offset + "-");
```

Comment:
To be clean you would need to set `Range` for non-zero offset

Reply:
Not sure what has to change, specifically; could you clarify what you'd like the `Range` to look like, in each of the scenarios under consideration? (`offset`=0 vs. `offset` > 0, with and without positive `length` argument, etc.).

Reply:
You would need to do below;
```java
if (length != null) {
  headerMap.put("Range", "bytes=" + offset + "-" + (offset + length -1));
  // Here offset is set anyway to satisfy range i.e. `Range: bytes=0-100` or `Range: bytes=10-100`
} else if offset > 0 {
  headerMap.put("Range", "bytes=" + offset + "-");
  // Here `Range` header is set only if offset > 0 i.e. for offset=10; `Range: bytes=10-`; for offset=0; nothing is set.
}
```

Reply:
Updated the client and test code as per ^. Thanks.